source/agora/consensus/validation/Transaction.d

```diff
@@ -94,10 +100,12 @@ public string isInvalidReason (
     }
 
     string isInvalidInput (in Input input, ref UTXO utxo_value,
-        ref Amount sum_unspent)
+        ref Amount sum_unspent, scope HasPenaltyDeposit hasPenaltyDeposit)
     {
         if (!findUTXO(input.utxo, utxo_value))
             return "Transaction: Input ref not in UTXO";
+        if (!sum_unspent.add(hasPenaltyDeposit(input.utxo)))
```

Comment:
If the input is a freeze output and was not slashed, we allow TX to spend the penalty deposit. But beware, we dont force them to spend it. if they dont add the penalty deposit to the outputs, that deposit is lost forever, essentially burned.

Reply:
Would it be fairer if it was refund by default?

Reply:
This was the simplest, yet works-perfectly-fine solution I could think of :) Since we probably want the refund to also be subject to the melting period we apply to frozen stakes, refunding them in a Coinbase TX was not my first choice.